src/templates/Front/index.js

```diff
@@ -0,0 +1,389 @@
+const matchTeaserType = teaserType =>
+  node => matchTeaser(node) && node.data.teaserType === teaserType
+
+const image = {
+  matchMdast: matchImageParagraph,
+  component: () => null,
+  isVoid: true
+}
+
+const extractImage = node => matchImageParagraph(node)
+  ? node.children[0].url
+  : undefined
+
+const styles = {
+  link: css({
+    color: 'inherit',
+    textDecoration: 'none'
+  })
+}
+
+const DefaultLink = ({ children }) => children
+
+const createSchema = ({
+  Link = DefaultLink
+} = {}) => {
+  const credit = {
+    matchMdast: matchParagraph,
+    component: ({ children, attributes }) =>
+      <TeaserFrontCredit attributes={attributes}>{children}</TeaserFrontCredit>,
+    editorModule: 'paragraph',
+    editorOptions: {
+      type: 'FRONTCREDIT',
+      placeholder: 'Credit'
+    },
+    rules: [
+      {
+        matchMdast: matchType('link'),
+        props: (node, index, parent, { ancestors }) => {
+          const teaser = ancestors.find(matchTeaser)
+          return {
+            title: node.title,
+            href: node.url,
+            color: teaser
+              ? teaser.data.linkColor
+              : colors.primary
+          }
+        },
+        component: ({ children, data, ...props }) =>
+          <Link href={props.href} passHref>
+            <TeaserFrontCreditLink {...props}>
+              {children}
+            </TeaserFrontCreditLink>
+          </Link>,
+        editorModule: 'link',
+        editorOptions: {
+          type: 'FRONTLINK'
+        }
+      }
+    ]
+  }
+
+  const title = (type, Headline) => ({
+    matchMdast: matchHeading(1),
+    component: ({ children, href, ...props }) => 
+      <Link href={href} passHref>
+        <a {...styles.link} href={href}>
+          <Headline {...props}>{children}</Headline>
+        </a>
+      </Link>,
+    props (node, index, parent, { ancestors }) {
+      const teaser = ancestors.find(matchTeaser)
+      return {
+        kind: parent.data.kind,
+        titleSize: parent.data.titleSize,
+        href: teaser
+          ? teaser.data.url
+          : undefined
+      }
+    },
+    editorModule: 'headline',
+    editorOptions: {
+      type,
+      placeholder: 'Titel',
+      depth: 1
+    }
+  })
+
+  const lead = {
+    matchMdast: matchHeading(4),
+    component: ({ children, attributes }) =>
+      <TeaserFrontLead attributes={attributes}>
+        {children}
+      </TeaserFrontLead>,
+    editorModule: 'headline',
+    editorOptions: {
+      type: 'FRONTLEAD',
+      placeholder: 'Lead',
+      depth: 4,
+      optional: true
+    }
+  }
+
+  const format = {
+    matchMdast: matchHeading(6),
+    component: ({ children, attributes }) =>
+      <Editorial.Format attributes={attributes}>
+        {children}
+      </Editorial.Format>,
+    editorModule: 'headline',
+    editorOptions: {
+      type: 'FRONTFORMAT',
+      placeholder: 'Format',
+      depth: 6,
+      optional: true
+    }
+  }
+
+  const frontImageTeaser = {
```

Comment:
Thanks for extracting these rules out of the otherwise huge schema object.

It might help readability to name these
frontImageTeaser*Rule*
frontSplitTeaser*Rule*
etc. to make clear at declaration time how they're gonna be used further down.